ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/MicrometerMetricsPrometheus.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.metrics.micrometer.prometheus
+
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.metrics.micrometer.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.util.pipeline.*
+import io.micrometer.core.instrument.distribution.*
+import io.micrometer.prometheus.*
+import io.prometheus.client.exporter.common.*
+
+class MicrometerMetricsPrometheus(config: Configuration) :
+    AbstractMicrometerMetrics<PrometheusMeterRegistry, MicrometerMetricsPrometheus.Configuration>(config) {
+
+    class Configuration : AbstractMicrometerMetrics.AbstractConfiguration<PrometheusMeterRegistry>() {
+
+        /**
+         * the meter registry
+         */
+        override var registry = PrometheusMeterRegistry(PrometheusConfig.DEFAULT)
+
+        override var distributionStatisticConfig: DistributionStatisticConfig = DistributionStatisticConfig.Builder()
+            .percentilesHistogram(true)
+            .build()
+
+        /**
+         * the path of the metrics endpoint
+         */
+        var endpointPath = "/metrics"
+
+        internal var authenticationConfig: Array<String?> = arrayOf()
+
+        /**
+         * enables authentication using the auth feature
+         */
+        fun authentication(authenticationConfig: Array<String?> = arrayOf(null)) {
+            this.authenticationConfig = authenticationConfig
+        }
+    }
+
+    companion object MetricsFeature :
+        AbstractMetricsFeature<PrometheusMeterRegistry, Configuration>(
+            ::MicrometerMetricsPrometheus,
+            MicrometerMetricsPrometheus::Configuration
+        ) {
+
+        override fun install(
+            pipeline: Application,
+            configure: Configuration.() -> Unit
+        ): AbstractMicrometerMetrics<PrometheusMeterRegistry, Configuration> {
+            val feature = super.install(pipeline, configure)
+
+            if (feature.config.authenticationConfig.isEmpty()) {
+                pipeline.routing {
+                    metricsEndpoint(feature.config)
+                }
+            } else {
+                pipeline.installMetricsEndpointWithAuth(feature.config)
+            }
+
+            return feature
+        }
+    }
+}
+
+
+
+/**
+ * the meter registry registered for this feature
+ */
+val PipelineContext<Unit, ApplicationCall>.meterRegistry
+    get() = call.application.feature(MicrometerMetricsPrometheus).config.registry
+
+
+internal fun Route.metricsEndpoint(config: MicrometerMetricsPrometheus.Configuration) =
+    get(config.endpointPath) {
+        prometheusMetrics()
+    }
+
+private suspend fun PipelineContext<Unit, ApplicationCall>.prometheusMetrics() {
+    call.respondText(contentType = ContentType.parse(TextFormat.CONTENT_TYPE_004)) {
```

Comment:
`CONTENT_TYPE_004` ?

Reply:
the value behind it is `text/plain; version=0.0.4; charset=utf-8`, I would take it as is so prometheus can easily parse it

Reply:
the value behind the constant is `text/plain; version=0.0.4; charset=utf-8` I would just use it like Prometheus needs it

Reply:
Can we reassign it to a val-property with self-explaining name?